crates/agglayer-aggregator-notifier/src/lib.rs

```
@@ -59,6 +65,45 @@ where
     }
 }
 
+impl Certifier for AggregatorNotifier<Certificate> {
+    type Input = Certificate;
+    type Proof = Proof;
+
+    fn certify(
+        &self,
+        local_state: LocalNetworkState,
+        certificate: Certificate,
+    ) -> Result<BoxFuture<Result<CertifierOutput<Self::Proof>, Error>>, Error> {
+        let proving_request = self.prover.prove(local_state.clone(), certificate.clone());
+
+        let mut state = local_state.clone();
+        let _native_outputs = state
+            .apply_certificate(&certificate)
+            .map_err(|_| Error::NativeExecutionFailed)?;
+
+        Ok(Box::pin(async move {
+            let proof = proving_request
+                .await
+                .map_err(|_| Error::ProofExecutionFailed)?;
```

The original error is lost with this `map_err`, can we log it to track them?